x-pack/plugins/stack_connectors/server/connector_types/stack/opsgenie/connector.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import crypto from 'crypto';
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import { AxiosError } from 'axios';
+import { CloseAlertParamsSchema, CreateAlertParamsSchema, Response } from './schema';
+import { CloseAlertParams, Config, CreateAlertParams, Secrets } from './types';
+import * as i18n from './translations';
+
+interface ErrorSchema {
+  message?: string;
+  errors?: {
+    message?: string;
+  };
+}
+
+export class OpsgenieConnector extends SubActionConnector<Config, Secrets> {
+  constructor(params: ServiceParams<Config, Secrets>) {
+    super(params);
+
+    this.registerSubAction({
+      method: this.createAlert.name,
+      name: 'createAlert',
+      schema: CreateAlertParamsSchema,
+    });
+
+    this.registerSubAction({
+      method: this.closeAlert.name,
+      name: 'closeAlert',
+      schema: CloseAlertParamsSchema,
+    });
+  }
+
+  public getResponseErrorMessage(error: AxiosError<ErrorSchema>) {
+    return `Message: ${
+      error.response?.data.errors?.message ?? error.response?.data.message ?? i18n.UNKNOWN_ERROR
```

Comment:
Can we add another fallback to include the error message? For example, `error.response?.data.errors?.message ?? error.response?.data.message ?? error.message ?? i18n.UNKNOWN_ERROR`. I am also thinking if we should improve the message thrown by the `SubActionConnector` class in the `request` method. At the moment we do

```
const errorMessage = this.getResponseErrorMessage(error);
throw new Error(errorMessage);
```

Maybe we can include more details like the status code. For example,

```
const errorMessage = `Status code: ${error.status}. Message: ${this.getResponseErrorMessage(error)}`;
throw new Error(errorMessage);
```

What do you think? @pmuellr Any information you think we should include in the error message to help us with SDHs?